hermes-consumers/src/main/java/pl/allegro/tech/hermes/consumers/consumer/result/undelivered/UndeliveredMessageHandlers.java

```diff
@@ -0,0 +1,33 @@
+package pl.allegro.tech.hermes.consumers.consumer.result.undelivered;
+
+import pl.allegro.tech.hermes.api.Subscription;
+import pl.allegro.tech.hermes.consumers.consumer.Message;
+import pl.allegro.tech.hermes.consumers.consumer.sender.MessageSendingResult;
+
+import javax.inject.Inject;
+import java.util.ArrayList;
+import java.util.List;
+
+public class UndeliveredMessageHandlers {
+
+    private final List<UndeliveredMessageHandler> undeliveredMessageHandlerList;
+
+    @Inject
+    public UndeliveredMessageHandlers(LogUndeliveredMessageHandler logUndeliveredMessageHandler) {
```

Comment:
why does it have to have at least once handler? also for safety you could move `new ArrayList()` to field definition so it's clear its always a new one